### PR TITLE
Initial workflow for Lookup Automox Host from Teams

### PR DESCRIPTION
**Pull request by ax-zyoutz**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [x] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

```
### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

<Insert Workflow Image Here>

#### Job

<Insert Job Image Here>

#### Artifact

<Insert Artifact Image Here>
```

### CODE REVIEW

**Comment by ax-zyoutz** [comment]
Screenshots of workflow and job output with a found and not found device:
Screen Shot 2022-02-09 at 8 33 14 PM
Screen Shot 2022-02-09 at 8 33 31 PM
Screen Shot 2022-02-09 at 8 33 46 PM

Teams threads:
Screen Shot 2022-02-09 at 8 34 42 PM
Screen Shot 2022-02-09 at 8 34 47 PM

File: workflows/Lookup_Automox_Host_From_Teams/help.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+* [Microsoft Teams](https://docs.rapid7.com/insightconnect/microsoft-teams/)
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. 
+Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+There are two workflow parameters you will need to configure in order to complete setup of your workflow:
```
Comment (ax-zyoutz) [inline workflows/Lookup_Automox_Host_From_Teams/help.md:26]:
```suggestion
There are three workflow parameters you will need to configure in order to complete setup of your workflow:
```

**Comment by ax-zyoutz** [comment]
@blaxminarayan-r7 Commit pushed and tested to refine the IPv4 regular expression with express number of digits

File: workflows/Lookup_Automox_Host_From_Teams/help.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+* Fetch host information from Automox quickly in Microsoft Teams
+* Determine when the system last checked in with the Automox platform
+* Quickly identify the number of upcoming patch count along with patch time 
+* Positively identify a system by looking up it up by IP address or hostname
+
+# Requirements
+
+* Automox API Key
+* InsightConnect License
```
Comment (mrinehart-r7) [inline workflows/Lookup_Automox_Host_From_Teams/help.md:16]:
```suggestion
* InsightConnect license
```

File: workflows/Lookup_Automox_Host_From_Teams/Lookup_Automox_Host_From_Teams.icon [hunk 3]
```diff
@@ -0,0 +1,2494 @@
...
+            "name": "Host Not Found Message",
+            "type": "action",
+            "plugin": {
+              "name": "Microsoft Teams",
+              "slugVendor": "rapid7",
+              "slugName": "microsoft_teams",
+              "slugVersion": "3.1.5",
+              "imageData": "https://us3.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/3.1.5/icon.png"
+            },
+            "identifier": "send_html_message",
```
Comment (ax-zyoutz) [inline workflows/Lookup_Automox_Host_From_Teams/Lookup_Automox_Host_From_Teams.icon:1496]: Changed to `send_html_message` here so we can include workflow job url the same as a host found message

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: workflows/Lookup_Automox_Host_From_Teams/help.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+# Key Features
+
+* Fetch host information from Automox quickly in Microsoft Teams
+* Determine when the system last checked in with the Automox platform
+* Quickly identify the number of upcoming patch count along with patch time 
+* Positively identify a system by looking up it up by IP address or hostname
+
+# Requirements
+
+* Automox API Key
```
Comment (mrinehart-r7) [inline workflows/Lookup_Automox_Host_From_Teams/help.md:15]:
```suggestion
* Automox API key
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by blaxminarayan-r7 — APPROVED** [review]

**Pull request merged**